app/controllers/api/v0/api_controller.rb

```diff
@@ -47,6 +47,17 @@ def authenticate!
         end
       end
 
+      def authenticate_super_admin!
```

Comment:
This feels like we're mixing authentication and authorization. Using a policy in `SiteConfigController` feels more in line with what we usually do.